sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala

```diff
@@ -2430,4 +2435,34 @@ class DDLParserSuite extends AnalysisTest {
     comparePlans(parsePlan(timestampTypeSql), insertPartitionPlan(timestamp))
     comparePlans(parsePlan(binaryTypeSql), insertPartitionPlan(binaryStr))
   }
+
+  test("as of syntax") {
+    var properties = new util.HashMap[String, String]
+    properties.put(TableCatalog.PROP_VERSION, "123456789")
+    comparePlans(
+      parsePlan("SELECT * FROM a.b.c VERSION AS OF 123456789"),
```

Comment:
shall we also test a bad `version` number?

Reply:
Since the wrong syntax (e.g. `VERSION AS OF -123456789`) will cause a parsing Exception for sure, I will skip a negative test here.